tests/e2e/config/bootstrap.js

```diff
@@ -122,13 +122,44 @@ function observeConsoleLogging() {
 	} );
 }
 
+/**
+ * Observe the given REST request.
+ */
+function observeRestRequest( req ) {
+	if ( req.url().match( 'wp-json' ) ) {
+		// eslint-disable-next-line no-console
+		console.log( '>>>', req.method(), req.url(), req.postData() );
+	}
+}
+
+/**
+ * Observe the given REST response.
+ */
+async function observeRestResponse( res ) {
+	if ( res.url().match( 'wp-json' ) ) {
+		const args = [ res.status(), res.request().method(), res.url() ];
+
+		// The response may fail to resolve if the test ends before it completes.
+		try {
+			args.push( await res.text() );
+		} catch ( err ) {} // eslint-disable-line no-empty
+
+		// eslint-disable-next-line no-console
+		console.log( ...args );
+	}
+}
+
 // Before every test suite run, delete all content created by the test. This ensures
 // other posts/comments/etc. aren't dirtying tests and tests don't depend on
 // each other's side-effects.
 beforeAll( async() => {
 	capturePageEventsForTearDown();
 	enablePageDialogAccept();
 	observeConsoleLogging();
+	if ( process.env.DEBUG_REST ) {
```

Comment:
@aaemnnosttv Can you elaborate a bit on what your changes here in this file try to solve? Seems like leftover debug stuff from development.

If you need to wait for a REST request to finish, [`page.waitForResponse`](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagewaitforresponseurlorpredicate-options) would be a good method to do so.

Reply:
This is development related, but has been critical in exposing the REST requests made that we need to be mocking. It's just for extra verbosity in the console when needed.

Reply:
Do you intend to remove this at some point, or just leave this in?

If we wanna keep this, it should be documented in the engineering handbook.

On the other hand, it seems way easier to debug request in the actual browser's Network tab instead of using `console.log` for the requests here...

Reply:
I agree that the browser's console is more robust and has more information but I don't believe we have a way of launching Puppeteer with Devtools enabled with the current config. Even if we could, you still have to manually enable "Preserve log" and configure the filtering you want, which is a pain when running these over and over.

This has been useful for me so far and I think it would be for others as well so I believe it's worth keeping. I'm fine with removing it at any point if we have something better though.

Reply:
As long as it's documented somewhere, I'm fine with it.

Reply:
Added a note to https://docs.google.com/document/d/1M1gUI0FKnP3RcbgMsU6YltEtNCHaH_9YGbH8F-7JefA/edit#